lib/event.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+import socket
+
+from utilities.toUTC import toUTC
+
+
+class Event(dict):
+
+    # We set default vaules so that we can later
+    # create an alert around these, and know when events
+    # have to use defaults
+    DEFAULT_STRING = 'None'
+
+    def add_required_fields(self):
+        if 'receivedtimestamp' not in self:
+            self['receivedtimestamp'] = toUTC(datetime.now()).isoformat()
+        if 'utctimestamp' not in self:
+            self['utctimestamp'] = toUTC(datetime.now()).isoformat()
+        if 'timestamp' not in self:
+            self['timestamp'] = toUTC(datetime.now()).isoformat()
+        if 'mozdefhostname' not in self:
+            self['mozdefhostname'] = socket.gethostname()
+        if 'tags' not in self:
+            self['tags'] = []
+        if 'category' not in self:
+            self['category'] = self.DEFAULT_STRING
+        if 'hostname' not in self:
+            self['hostname'] = self.DEFAULT_STRING
+        if 'processid' not in self:
+            self['processid'] = self.DEFAULT_STRING
+        if 'processname' not in self:
+            self['processname'] = self.DEFAULT_STRING
+        if 'severity' not in self:
+            self['severity'] = self.DEFAULT_STRING
```

Comment:
I wasn't 100% sure here if we wanted to set the severity to 'INFO', or a legitimate default value, or just use the 'None' placeholder here.

Reply:
I'd say we want it to be a unique default so we can still tell if there are events that do not contain the required fields.

Reply:
UNDEFINED so we can query it later?

Reply:
I don't think any events will have 'None' as the value, so we should be able to search for these events. I'm fine with switching the placeholder text to be whatever, if folks think 'None' isn't unique enough to search on.

Reply:
I'd suggest UNKNOWN in caps as it should be obvious and easy to search.

Reply:
I agree with @jeffbryner